mtags/src/main/scala-2/scala/meta/internal/pc/MetalsGlobal.scala

```diff
@@ -287,22 +287,36 @@ class MetalsGlobal(
             }
           }
         case ThisType(sym) =>
-          if (history.tryShortenName(name)) NoPrefix
+          val owners = sym.ownerChain
+          // to make sure we always use renamed package
+          // what is saved in renames is actually companion module of a package
+          val renamedOwnerIndex =
```

Comment:
So this turned out harder than anticipated. We would not properly rename packages before, which would also pop up in signatures.

Reply:
huh, interesting. Great job finding this!

Reply:
I have no idea actually why the package type and module type is different, but I don't think this will actually break anything. If it will, I will probably need to figure it out :S